tests/dataset_optimization_shared.py

```diff
@@ -8,7 +9,43 @@
 logger = get_logger(__name__)
 
 
-def cleanup(test_dataset: OptimizationDataset):
+def get_unique_id():
+    return os.getenv("UNIQUE_ID", "").replace(".", "-").replace("/", "-")
+
+
+def cleanup_conflict_by_unique_id(unique_id: typing.Optional[str]):
+    if unique_id:
+        conflicting_git_repo_ids = [
+            git_repo["id"]
+            for git_repo in GitRepo.list()
+            if unique_id in git_repo["name"]
```

Comment:
why `in`?

Reply:
To cleanup items which may be with a different name, but containing this exactly string. This is a failsafe